File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.py
class Solution:
    def closestTarget(self, words: List[str], target: str, startIndex: int) -> int:
        start = words[startIndex]
        n = len(words)
        l_steps = 0
        r_steps = 0
        if words[startIndex] == target: return 0
        for i in range(1, max(abs(startIndex - n), abs(startIndex))):
            l = words[(startIndex - i ) % n]
            r = words[(startIndex + i) % n] 
            l_steps += 1
            r_steps += 1

            if l == target or r == target:
                return min(l_steps, r_steps)
        return -1

File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.cpp
using namespace std;

class Solution {
public:
    int closestTarget(vector<string>& words, string target, int startIndex) {
        int n = words.size();
        
        if (words[startIndex] == target) {
            return 0;
        }

        for (int i = 1; i < n; ++i) {
            string l = words[(startIndex - i + n) % n];
            string r = words[(startIndex + i) % n];

            if (l == target || r == target) {
                return i;
            }
        }
        return -1;
    }
};